crates/bevy_ecs/macros/src/component.rs

```diff
@@ -65,15 +69,28 @@ pub fn derive_component(input: TokenStream) -> TokenStream {
     TokenStream::from(quote! {
         impl #impl_generics #bevy_ecs_path::component::Component for #struct_name #type_generics #where_clause {
             const STORAGE_TYPE: #bevy_ecs_path::component::StorageType = #storage;
+
+            #[allow(unused_variables)]
+            fn register_component_hooks(hooks: &mut #bevy_ecs_path::component::ComponentHooks) {
+                #on_add
+                #on_insert
+                #on_remove
+            }
         }
     })
 }
 
 pub const COMPONENT: &str = "component";
 pub const STORAGE: &str = "storage";
+pub const ON_ADD: &str = "on_add";
+pub const ON_INSERT: &str = "on_insert";
+pub const ON_REMOVE: &str = "on_remove";
 
 struct Attrs {
     storage: StorageTy,
+    on_add: Option<Meta>,
```

Comment:
We should probably change these to `Option<ExprPath>` so that the derive macro can give a bit more help when given something unexpected (also helps self-document).

Unless there's a reason we should parse `Meta` instead?